### PR TITLE
DM-18393: Add some needed features for testing at SLAC

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: include/lsst/afw/cameraGeom/Detector.h [hunk 1]
```diff
@@ -89,21 +90,23 @@ class Detector final : public table::io::PersistableFacade<Detector>, public tab
              lsst::geom::Box2I const &bbox, lsst::afw::table::AmpInfoCatalog const &ampInfoCatalog,
              Orientation const &orientation, lsst::geom::Extent2D const &pixelSize,
              TransformMap::Transforms const &transforms,
-             CrosstalkMatrix const &crosstalk = CrosstalkMatrix());
+             CrosstalkMatrix const &crosstalk = CrosstalkMatrix(),
+             std::string const &physicalType="unknown");
```
Comment (TallJimbo) [inline include/lsst/afw/cameraGeom/Detector.h]: What do you think about just using an empty string to mean "unknown"?  It's an easier sentinel value to remember.

File: src/cameraGeom/Detector.cc [hunk 2]
```diff
@@ -212,7 +243,7 @@ class DetectorFactory : public table::io::PersistableFactory {
 
     std::shared_ptr<table::io::Persistable> read(InputArchive const& archive,
                                                  CatalogVector const& catalogs) const override {
-        auto const & keys = PersistenceHelper::get();
+        auto const & keys = PersistenceHelper(catalogs.front().getSchema());
```
Comment (TallJimbo) [inline src/cameraGeom/Detector.cc]: While C++ lets you get away with binding a temporary to a const reference, for a local variable like this I think it's better to just assign it to a value (i.e. `auto const` instead of `auto const &`).

File: tests/test_detector.py [hunk 3]
```diff
@@ -84,7 +86,7 @@ def duplicateAmpName(dw):
             DetectorWrapper(modFunc=duplicateAmpName)
 
         # These break in the pybind layer
-        for crosstalk in ([1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0],  # 1D and not numpy
+        for crosstalk in ([11.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0],  # 1D and not numpy
```
Comment (TallJimbo) [inline tests/test_detector.py]: Was this change intentional?

Comment (RobertLuptonTheGood) [reply tests/test_detector.py]: No; used in setting a breakpoint when trying to find the numpy problem.  I'll revert it.

File: src/cameraGeom/Detector.cc (hunk 2 above)
Comment (RobertLuptonTheGood) [reply src/cameraGeom/Detector.cc]: Cargo-culted from the previous lone, which worked because it was a singleton.  I'll change it

File: include/lsst/afw/cameraGeom/Detector.h (hunk 1 above)
Comment (RobertLuptonTheGood) [reply include/lsst/afw/cameraGeom/Detector.h]: Should be OK in C++; yamlCamera can still assign "unknown" which I prefer in the end
